app/config/database.js

```diff
@@ -0,0 +1,46 @@
+const { DefaultAzureCredential } = require('@azure/identity')
+
+function isProd () {
+  return process.env.NODE_ENV === 'production'
+}
+
+const hooks = {
+  beforeConnect: async (cfg) => {
+    if (isProd()) {
+      const credential = new DefaultAzureCredential()
+      const accessToken = await credential.getToken('https://ossrdbms-aad.database.windows.net', { requestOptions: { timeout: 1000 } })
+      cfg.password = accessToken.token
```

Comment:
If the image is the production definition, ie not running locally.  Then should use the Azure Managed Identity for connectivity.

The way this works for Azure Postgres requires devs to include a bit more setup than some of the other PaaS products like Azure Service Bus.

If the managed identity is being used, we can use the `DefaultAzureCredential` which will seek out credentials to use in a priority order.  One of those being the presence of a Managed Identity token.  (Pod Identity running in the cluster will make this token available to the pod).

We then use that token to call a fixed Azure endpoint to get another token for our connection string.  We set that token we get back as the password in the connection string.

We're adding a timeout of 1 second as sometimes that process takes longer than the default half a second to complete.  It would automatically try again because of the retry logic specified below.  But always good to try to minimise unnecessary errors.

It can still take longer than a second but that's much rarer.